packages/widget.vite-plugin/src/plugin.ts

```diff
@@ -572,3 +589,107 @@ function extractWidgetConfig(objectExpression: ObjectExpression) {
 
   return JSON.parse(widgetConfigString);
 }
+
+function setWidgetSettings(
+  widgetRid: string,
+  foundryUrl: URL,
+  localhostUrl: string,
+  entrypointToJsSourceFileMap: Record<string, Set<string>>,
+  entrypointFileName: string,
+) {
+  const widgetDevModeSettings = {
+    entrypointJs: [
+      `/${VITE_INJECTIONS}`,
+      "/@vite/client",
+      ...entrypointToJsSourceFileMap[entrypointFileName],
+    ].map((file) => ({
+      filePath: `${localhostUrl}${file}`,
+      scriptType: { type: "module", module: {} },
+    })),
+    entrypointCss: [],
+  };
+  return fetch(
+    `${foundryUrl.origin}/view-registry/api/dev-mode/settings/${widgetRid}`,
```

Comment:
How are you thinking about this as public API long term? On the deploy side we have gateway APIs but for Artifacts we don't for direct npm access etc.

Reply:
We should align behind public API equivalents of these once we have them!